equicord: unable to use with current discord packages due to a different location of the patcher.js file

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Equicord package can't be used as a drop-in replacement for the Vencord package in the pkgs.discord.override. 

### Steps to reproduce

1. Try add discord package like this:
```nix
{pkgs, ...}: {
    environment.systemPackages = with pkgs; [
        (discord.override {
             withVencord = true;
             vencord = equicord;
        })
    ];
}
```
2. Launch discord in the console
3. Observe the error (see the logs)

### Expected behaviour

Discord successfully launches with Equicord installed

### Screenshots

Equicord package structure:

![Image](https://github.com/user-attachments/assets/3352b406-f1a6-4bf6-be5f-f9a31293b587)

Vencord package structure:

![Image](https://github.com/user-attachments/assets/712e4fbe-cd0d-4942-a98b-c8e7a70acc79)

### Relevant log output

```console
discord
[Nix] Disabling updates already done
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Cannot find module '/nix/store/b1mqmf9vsgyzmq47rw6d3grmxz03x8bc-equicord-1.11.4/patcher.js'
Require stack:
- /nix/store/n425h53c0wiijzlqkzr0nnk5kpcnqcsy-discord-0.0.85/opt/Discord/resources/app.asar/index.js
-
    at Module._resolveFilename (node:internal/modules/cjs/loader:1232:15)
    at s._resolveFilename (node:electron/js2c/browser_init:2:124479)
    at Module._load (node:internal/modules/cjs/loader:1058:27)
    at c._load (node:electron/js2c/node_init:2:16955)
    at Module.require (node:internal/modules/cjs/loader:1318:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/nix/store/n425h53c0wiijzlqkzr0nnk5kpcnqcsy-discord-0.0.85/opt/Discord/resources/app.asar/index.js:1:1)
    at Module._compile (node:internal/modules/cjs/loader:1484:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1564:10)
    at Module.load (node:internal/modules/cjs/loader:1295:32)
```

### Additional context

I think there's 2 ways how it can be fixed:
1. Change Equicord package (make structure more like Vencord's one)
2. Change Discord package (add Eqicord support)

I can make a PR with the fix.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.3, NixOS, 25.05 (Warbler), 25.05.20250218.73cf49b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/kdynjy1mbgkdg4p196v9gx6ljpf7q4nk-source`

### Notify maintainers

@NotAShelf 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/networking/instant-messengers/discord/darwin.nix
+++ b/pkgs/applications/networking/instant-messengers/discord/darwin.nix
@@ -17,6 +17,8 @@
   openasar,
   withVencord ? false,
   vencord,
+  withEquicord ? false,
+  equicord,
   withMoonlight ? false,
   moonlight,
   commandLineArgs ? "",
@@ -81,7 +83,12 @@ stdenv.mkDerivation {
       echo '{"name":"discord","main":"index.js"}' > $out/Applications/${desktopName}.app/Contents/Resources/app.asar/package.json
       echo 'require("${vencord}/patcher.js")' > $out/Applications/${desktopName}.app/Contents/Resources/app.asar/index.js
     ''
-
+    + lib.strings.optionalString withEquicord ''
+      mv $out/Applications/${desktopName}.app/Contents/Resources/app.asar $out/Applications/${desktopName}.app/Contents/Resources/_app.asar
+      mkdir $out/Applications/${desktopName}.app/Contents/Resources/app.asar
+      echo '{"name":"discord","main":"index.js"}' > $out/Applications/${desktopName}.app/Contents/Resources/app.asar/package.json
+      echo 'require("${equicord}/patcher.js")' > $out/Applications/${desktopName}.app/Contents/Resources/app.asar/index.js
+    ''
     + lib.strings.optionalString withMoonlight ''
       mv $out/Applications/${desktopName}.app/Contents/Resources/app.asar $out/Applications/${desktopName}.app/Contents/Resources/_app.asar
       mkdir $out/Applications/${desktopName}.app/Contents/Resources/app.asar

--- a/pkgs/applications/networking/instant-messengers/discord/darwin.nix
+++ b/pkgs/applications/networking/instant-messengers/discord/darwin.nix
@@ -24,11 +24,14 @@
   commandLineArgs ? "",
 }:
 
-assert lib.assertMsg (
-  !(withMoonlight && withVencord)
-) "discord: Moonlight and Vencord can not be enabled at the same time";
-
 let
+  discordMods = [
+    withVencord
+    withEquicord
+    withMoonlight
+  ];
+  enabledDiscordModsCount = builtins.length (lib.filter (x: x) discordMods);
+
   disableBreakingUpdates =
     runCommand "disable-breaking-updates.py"
       {
@@ -43,6 +46,9 @@ let
         chmod +x $out/bin/disable-breaking-updates.py
       '';
 in
+assert lib.assertMsg (
+  enabledDiscordModsCount <= 1
+) "discord: Only one of Vencord, Equicord or Moonlight can be enabled at the same time";
 stdenv.mkDerivation {
   inherit
     pname

--- a/pkgs/applications/networking/instant-messengers/discord/linux.nix
+++ b/pkgs/applications/networking/instant-messengers/discord/linux.nix
@@ -73,10 +73,15 @@
   disableUpdates ? true,
   commandLineArgs ? "",
 }:
-assert lib.assertMsg (
-  !(withMoonlight && withVencord)
-) "discord: Moonlight and Vencord can not be enabled at the same time";
+
 let
+  discordMods = [
+    withVencord
+    withEquicord
+    withMoonlight
+  ];
+  enabledDiscordModsCount = builtins.length (lib.filter (x: x) discordMods);
+
   disableBreakingUpdates =
     runCommand "disable-breaking-updates.py"
       {
@@ -91,6 +96,9 @@ let
         chmod +x $out/bin/disable-breaking-updates.py
       '';
 in
+assert lib.assertMsg (
+  enabledDiscordModsCount <= 1
+) "discord: Only one of Vencord, Equicord or Moonlight can be enabled at the same time";
 stdenv.mkDerivation rec {
   inherit
     pname

--- a/pkgs/applications/networking/instant-messengers/discord/linux.nix
+++ b/pkgs/applications/networking/instant-messengers/discord/linux.nix
@@ -61,6 +61,8 @@
   openasar,
   withVencord ? false,
   vencord,
+  withEquicord ? false,
+  equicord,
   withMoonlight ? false,
   moonlight,
   withTTS ? true,
@@ -207,6 +209,12 @@ stdenv.mkDerivation rec {
       mkdir $out/opt/${binaryName}/resources/app.asar
       echo '{"name":"discord","main":"index.js"}' > $out/opt/${binaryName}/resources/app.asar/package.json
       echo 'require("${vencord}/patcher.js")' > $out/opt/${binaryName}/resources/app.asar/index.js
+    ''
+    + lib.strings.optionalString withEquicord ''
+      mv $out/opt/${binaryName}/resources/app.asar $out/opt/${binaryName}/resources/_app.asar
+      mkdir $out/opt/${binaryName}/resources/app.asar
+      echo '{"name":"discord","main":"index.js"}' > $out/opt/${binaryName}/resources/app.asar/package.json
+      echo 'require("${equicord}/desktop/patcher.js")' > $out/opt/${binaryName}/resources/app.asar/index.js
     ''
     + lib.strings.optionalString withMoonlight ''
       mv $out/opt/${binaryName}/resources/app.asar $out/opt/${binaryName}/resources/_app.asar

--- a/pkgs/by-name/eq/equicord/package.nix
+++ b/pkgs/by-name/eq/equicord/package.nix
@@ -3,7 +3,7 @@
   git,
   lib,
   nodejs,
-  pnpm_9,
+  pnpm_10,
   stdenv,
   nix-update-script,
   buildWebExtension ? false,
@@ -14,25 +14,25 @@ stdenv.mkDerivation (finalAttrs: {
   # the Equicord repository. Dates as tags (and automatic releases) were the compromise
   # we came to with upstream. Please do not change the version schema (e.g., to semver)
   # unless upstream changes the tag schema from dates.
-  version = "2025-07-25";
+  version = "2025-08-24";
 
   src = fetchFromGitHub {
     owner = "Equicord";
     repo = "Equicord";
     tag = "${finalAttrs.version}";
-    hash = "sha256-6p7lfdRaxQusqZf86a1qclal2E0LusYYT1y3BPkOSKY=";
+    hash = "sha256-BK0Mvy0Bp0Q6wj4aECEtyGsW56hqbLkkELZ9yN+QRw8=";
   };
 
-  pnpmDeps = pnpm_9.fetchDeps {
+  pnpmDeps = pnpm_10.fetchDeps {
     inherit (finalAttrs) pname version src;
     fetcherVersion = 1;
-    hash = "sha256-LXbr/qh1QZXQ/Na5180Im2WW8KbWU78Oo0O12SiNMWg=";
+    hash = "sha256-xVnryPA7+gnRvpMzuFJl4YeEPOky2+iOu76V3Rf6bow=";
   };
 
   nativeBuildInputs = [
     git
     nodejs
-    pnpm_9.configHook
+    pnpm_10.configHook
   ];
 
   env = {

